internal/contracts/contracts.go

```diff
@@ -274,4 +275,31 @@ func (mc *JSONContract) Unmarshal() (Contract, error) {
 		mc.StateNonce,
 	}
 	return c, nil
+	
+}
+
+
+
+func GenerateRandomContract() Contract{
+	b := rand.Read(make([]byte, 32))
+	
+	genSenderPubKey := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
```

Comment:
- [ ] `ecdsa.GenerateKey` generates a private key. You need to extract the public key from the private key

How this is done: https://github.com/SIGBlockchain/project_aurum/blob/master/internal/publickey/publickey_test.go#L56